### ISSUE TITLE
[Edge case] Included build has same name as a project of the main build

### ISSUE THREAD

**Issue by tomnis** [ADTRAN/gradle-scala-multiversion-plugin#27]
Hi folks,

Thanks for this plugin, we have found it really useful.

Recently we refactored one of our gradle projects to split into 2 subprojects. When we run tasks with `-PallScalaVersions`, we get an error like:

```
$ gw clean compileScala -PallScalaVersions
...
Execution failed for task ':recurseWithScalaVersion_2.12.10'.
> Included build in /Users/tomnis/foo has name 'foo' which is the same as a project of the main build.
```

We are not using composite builds, and my understanding is this plugin does not use composite builds either, but `GradleBuild` uses a similar mechanism.

One of our subprojects has the same name as our repository, but we override `rootProject.name` in `settings.gradle`.
We are using gradle 6.5.1 and multiversion 1.0.35.

Based on some further reading I believe the root cause is a regression in gradle 6.x. I think we are running into a combination of these 2 issues:

https://github.com/gradle/gradle/issues/12872
https://github.com/gradle/gradle/issues/3828 (we set `rootProject.name` in our `settings.gradle`, but that appears to not take effect for dynamic recursion tasks)

In issue 12872, @/ark100 suggested a workaround of setting `buildName` on dynamic `GradleBuild` tasks:

```
task task1(type: GradleBuild) {
    buildFile = file('gen/build.gradle')
    // ...
    if(6 <= gradle.gradleVersion.split(Pattern.quote('.'))[0].toInteger() ){
        buildName = "t1"
    }
}
```

I modified `ScalaMultiVersionPlugin.groovy` and published a local snapshot, this workaround seems to work for us.
https://github.com/tomnis/gradle-scala-multiversion-plugin/commit/a21ba5ac0c8c350485561608927094ff967f9096

I am unclear whether this repo is still being maintained, but just wanted y'all to be aware of this edge case and put this out there for anyone else running into the same issue. Thanks again for this plugin, it has been really handy for us.

**Comment by nathanalderson** [comment]
I don't have much time for development, but I'm happy to accept a pull request. What you describe seems reasonable.

**Comment by tomnis** [comment]
Thanks for the quick response @nathanalderson, I will open a pr

**Comment by dyno** [comment]
still have this problem in `com.adtran:scala-multiversion-plugin:1.0.36`
and my gradle version is 6.8.3

**Comment by tomnis** [comment]
hi @dyno we actually started running into this again too, trying to cross-compile for 2.13 in addition to 2.11 and 2.12 that we already had working. 

multiversion-plugin 1.0.36, gradle 6.8.3.

I modified `ScalaMultiVersionPlugin.groovy` to further "uniquify"  dynamic buildName (included scala version):

```
                        // workaround name clash issue in Gradle 6+. See:
                        // https://github.com/ADTRAN/gradle-scala-multiversion-plugin/issues/27.
                        if (6 <= project.gradle.gradleVersion.split(Pattern.quote('.'))[0].toInteger() ){
                            buildName = project.name + "_" + ver
                        }
```

Published a local plugin snapshot and this seems to work for cross-compiling for 3 scala versions.

https://github.com/tomnis/gradle-scala-multiversion-plugin/blob/ca0e276acf612028b580ec7ec69211cee12beb5e/src/main/groovy/com/adtran/ScalaMultiVersionPlugin.groovy#L138
